web/src/main/java/org/mskcc/cbio/oncokb/controller/advice/ExceptionTranslator.java

```diff
@@ -0,0 +1,48 @@
+package org.mskcc.cbio.oncokb.controller.advice;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.http.HttpStatus;
+
+/**
+ * Define your global exception handlers
+ */
+
+@ControllerAdvice
+public class ExceptionTranslator {
+    @ExceptionHandler(ApiHttpErrorException.class)
+    public final ResponseEntity<ApiHttpError> handleException(ApiHttpErrorException ex, WebRequest request) {
+        String path = request.getDescription(false).replaceFirst("uri=", "");
+        ApiHttpError error = new ApiHttpError(path, ex.getHttpStatus(), ex.getMessage());
+        return new ResponseEntity<>(error, ex.getHttpStatus());
+    }
+
+    @ExceptionHandler(org.oncokb.oncokb_transcript.ApiException.class)
+    public final ResponseEntity<ApiHttpError> handleTranscriptApiException(org.oncokb.oncokb_transcript.ApiException ex, WebRequest request) {
+        String path = request.getDescription(false).replaceFirst("uri=", "");
+        HttpStatus httpStatus = HttpStatus.INTERNAL_SERVER_ERROR;
+        try {
+            httpStatus = HttpStatus.valueOf(ex.getCode());
+        } catch (IllegalArgumentException e) {}
```

Comment:
Do we have the ability to log this case? Would it expected to enter this catch block?

Reply:
It will enter this catch block when `ex.getCode() == 0`. 0 is the default status code for ApiException classes and not a valid http status code, so the default 500 error is thrown instead.

Reply:
Logging is definitely possible, but sometimes the http code is optional. Maybe a comment here explaining the decision would suffice? What do you think @jfkonecn?

Reply:
yeah I'm good with comment. An empty catch block always makes me think something is wrong.